handler.c

```diff
@@ -2171,11 +2171,18 @@ setup_keys(void)
         if (!auth_info[i].data) {
             WARN("Cannot setup %s because auth data is missing!\n",
                  auth_info[i].pretty_name);
+
+            if (auth_info[i].name == EFI_IMAGE_SECURITY_DATABASE1) {
```

Comment:
I'm not familiar with this code but I am receiving conflicting messages from this code: it looks like `auth_info[i].data` is not optional based on the warning but somehow then it is optional? If that is the case, I believe the whole check should look different.

Reply:
No dbx is not fatal, but may be an issue if it's not on purpose. So a warning saying that we could not setup dbx might remain relevant. Otherwise, I can remove it in the case of the optional dbx. I don't have a personal preference about this.

Reply:
Instead of special casing `EFI_IMAGE_SECURITY_DATABASE1` here, can you add a `bool required` to `struct auth_info` and check for that instead?

Reply:
Yes. I'll do this.

Reply:
Done.